x/models.py

```diff
@@ -45,7 +47,7 @@ def max_values(self, observation, train=False):
     def policy(self, observation, train=False):
         raise NotImplementedError
 
-    def update(self, inputs, targets):
+    def update(self, inputs, targets, actions):
```

Comment:
Most update methods will require the action taken (table does) to update its values. Even if Keras doesn't.